### DISCUSSION TITLE
How are Bootstrap styles generated using Sass?

### DISCUSSION THREAD

**Discussion by jsphkm** [twbs/bootstrap#31150] [General] [answered]
I downloaded the npm bootstrap package, went into `node_modules/bootstrap/dist/css/bootstrap.css`, and looked at the styles for `.alert-danger`:
```
.alert-danger {
  color: #721c24;
  background-color: #f8d7da;
  border-color: #f5c6cb;
}
```
When I search those color codes, they do not appear in any other files. Where are these codes being defined/calculated? I’m interested in customizing those values using Sass, and I’m trying to keep specificity overrides as my last resort.

**Comment by patrickhlauke** [comment, marked as answer]
https://getbootstrap.com/docs/4.5/getting-started/theming/#variable-defaults

https://github.com/twbs/bootstrap/blob/v4-dev/scss/_variables.scss
